### PR TITLE
Removed guessing game from automated deployment.

### PR DESCRIPTION
**Pull request by jamescarlyle**
### Why is this change needed?
https://github.com/obscuronet/obscuro-internal/issues/770

### What changes were made as part of this PR:

Remove all references to the Guessing Game in go-obscuro.
Switched the contract deployer test to use ERC20 contract instead.

### :rotating_light: Definition of done :rotating_light:
- [ Yes] Unit tests added to cover new or changed functionality (test updated to use ERC20)
- [ Yes] Docs pages updated to cover new or changed functionality (existing docs updated to point to new Guessing Game URL)
- [Yes ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: docs/community/journey.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ The guessing game is a very basic contract that showcases some of Obscuro's capa
 
 Before we get into how the game works, let's see it in action and play. See the instructions [here](https://docs.obscu.ro/testnet/example-dapps.html)
 
-Now that we've played the game let's try deploying the Guessing game to the Obscuro testnet. You can clone the repo from [here](https://github.com/obscuronet/number-guessing-game). 
+Now that we've played the game let's try deploying the Guessing game to the Obscuro testnet. You can clone the repo from [here](https://github.com/obscuronet/sample-applications/number-guessing-game).
```
Comment (joeldudley) [inline docs/community/journey.md:49]: This link is broken. It would be https://github.com/obscuronet/sample-applications/tree/main/number-guessing-game.

Comment (joeldudley) [reply docs/community/journey.md:49]: And you can't clone a subfolder like that, AFAIK. You have to clone the entire `sample-applications`.

File: docs/testnet/example-dapps.md [hunk 2]
```diff
@@ -2,44 +2,37 @@
...
+1. The game should show you two messages, one confirming the success of the Approval transaction, and the second confirming the result of the guess attempt (whether correct, wrong, warmer or colder than the last guess).
 
 Once the guess transaction is completed you can check the guess transaction on ObscuroScan:
 1. In MetaMask click on the transaction to open it then click `Copy Transaction ID`. Open [ObscuroScan](http://testnet.obscuroscan.io/).
 1. Paste your copied transaction ID into the search box to find your individual guess transaction. Note how the transaction data visible to everyone is encrypted in the field `EncryptedTxBlob` and how (for now) the transaction is decrypted to allow developers to confirm correct behaviour of the network.
 1. You can see your guess as the number at the right hand end of the `input` value in a hexadecimal format, e.g. a guess of 99 is shown as 63.
 
 ### Known Issues and Limitations
-1. You requested game tokens using the Discord thread but there was no response.
+1. You have requested OBX tokens using the Discord thread but there was no response.
```
Comment (joeldudley) [inline docs/testnet/example-dapps.md:30]: Is this still relevant? Aren't you saying above that tokens are auto-allocated?

**Review by joeldudley — CHANGES_REQUESTED** [review]
A few requested changes.

File: docs/community/journey.md (hunk 1 above)
Comment (jamescarlyle) [reply docs/community/journey.md:49]: doh.. yes

File: docs/testnet/example-dapps.md (hunk 2 above)
Comment (jamescarlyle) [reply docs/testnet/example-dapps.md:30]: Yep

**Review by joeldudley — APPROVED** [review]
LGTM

**Pull request merged**